pkg/operations/kubernetesupgrade/upgradeagentnode.go

```diff
@@ -112,30 +113,27 @@ func (kan *UpgradeAgentNode) Validate(vmName *string) error {
 		return err
 	}
 
-	ch := make(chan struct{}, 1)
-	go func() {
-		for {
+	retryTimer := time.NewTimer(time.Millisecond)
+	timeoutTimer := time.NewTimer(timeout)
+	for {
+		select {
+		case <-timeoutTimer.C:
+			retryTimer.Stop()
+			kan.logger.Errorf("Node was not ready within %v", timeout)
+			return fmt.Errorf("Node was not ready within %v", timeout)
```

Comment:
[nit] err := fmt.Errorf("Node was not ready within %v", timeout")
kan.logger.Error(err.Error())
return err